examples/seismic/acoustic/wavesolver.py

```diff
@@ -90,30 +91,38 @@ def forward(self, src=None, rec=None, u=None, model=None, save=None, **kwargs):
             The time-constant velocity.
         save : bool, optional
             Whether or not to save the entire (unrolled) wavefield.
+        factor : int, optional
+        Downsampling factor to save snapshots of the wavefield.
 
         Returns
         -------
         Receiver, wavefield and performance summary
         """
         # Source term is read-only, so re-use the default
         src = src or self.geometry.src
-        # Create a new receiver object to store the result
+        # Create a new receiver object to store the result 
         rec = rec or self.geometry.rec
 
         # Create the forward wavefield if not provided
         u = u or TimeFunction(name='u', grid=self.model.grid,
-                              save=self.geometry.nt if save else None,
-                              time_order=2, space_order=self.space_order)
-
+                                save=self.geometry.nt if save and factor is None else None,
+                                time_order=2, space_order=self.space_order)
+        if factor: # Create snapshots of the forward wavefield
+            usnaps = create_snapshot_time_function(self.model, 'usnaps', self.geometry, self.space_order, factor)
```

Comment:
variable definitions should not be conditional-dependent

Reply:
Now `usnaps` is defined unconditionally.